middle-ram.hpp

```diff
@@ -61,15 +61,19 @@ struct middle_ram_t : public middle_t {
     struct ramWay {
         taglist_t tags;
         idlist_t ndids;
+
+        ramWay(const taglist_t &t, const idlist_t &n) : tags(t), ndids(n) {}
     };
 
     struct ramRel {
         taglist_t tags;
         memberlist_t members;
+
+        ramRel(const taglist_t &t, const memberlist_t &m) : tags(t), members(m) {}
     };
 
-    std::vector<std::vector<ramWay> > ways;
-    std::vector<std::vector<ramRel> > rels;
+    std::vector<std::unique_ptr<std::vector<std::unique_ptr<ramWay> > > > ways;
```

Comment:
this is a vector of pointers to vectors of pointers to ways?

I'm wondering if it makes sense to typedef `std::vector<std::unique_ptr<ramWay> >` for clarity that it's a block of ways